pypeit/spectrographs/p200_ngps.py

```diff
@@ -0,0 +1,619 @@
+
+        
+        msgs.warn('Cannot determine if frames are of type {0}.'.format(ftype))
+        return np.zeros(len(fitstbl), dtype=bool)
+
+
+class P200NGPSSpectrograph_R(P200NGPSSpectrograph):
+    """
+    Child to handle P200/NGPS R-Channel specific code
+    """
+    name = 'p200_ngps_R'
+    camera = 'NGPS_R'
+    header_name = 'NGPS_R'
+    supported = True
+    comment = 'R-Channel'
+
+    def get_rawimage(self, raw_file, det):
+        """
+        Read raw spectrograph image files and return data and relevant metadata
+        needed for image processing.
+
+        For P200/NGPS, the ``DATASEC`` and ``OSCANSEC`` regions are read
+        directly from the file header and are automatically adjusted to account
+        for the on-chip binning.  This is a simple wrapper for
+        :func:`pypeit.spectrographs.spectrograph.Spectrograph.get_rawimage` that
+        sets ``sec_includes_binning`` to True.  See the base-class function for
+        the detailed descriptions of the input parameters and returned objects.
+        """
+
+        return super().get_rawimage(raw_file, det=1, sec_includes_binning=True)
+    
+
+
+    def compound_meta(self, headarr: List[fits.Header], meta_key: str):
+        """
+        Methods to generate metadata requiring interpretation of the header
+        data, instead of simply reading the value of a header card.
+
+        Args:
+            headarr (:obj:`list`):
+                List of `astropy.io.fits.Header`_ objects.
+            meta_key (:obj:`str`):
+                Metadata keyword to construct.
+
+        Returns:
+            object: Metadata value read from the header(s).
+        """
+        # Handle dispangle and mjd from superclass method
+        retval = super().compound_meta(headarr, meta_key)
+        
+        # If superclass could not handle the meta key
+        if retval is not None:
+            return retval
+        
+        if meta_key == 'mjd':
+            return Time(headarr[0]['UTSHUT']).mjd
+        elif meta_key == 'dispangle':
+            try:
+                return 0
+            except Exception as e:
+                msgs.warn("Could not read dispangle from header:" + msgs.newline() + str(headarr[0]['ANGLE']))
+                raise e    
+            
+        elif meta_key == 'binning':
+            # Always the same binning for DET01 and DET02    
+            binspat = headarr[1]['BINSPAT'] 
+            binspec = headarr[1]['BINSPEC']
+            return f"{binspec},{binspat}"
+        
+        # If there is no target keyword, return image type
+        elif meta_key == 'target':
+            if headarr[0]['TARGET'] is not None:
+                return headarr[0]['TARGET']
+            else:
+                return headarr[0]['IMGTYPE']
+        elif meta_key == 'dichroic': 
+            return None
+        else:
+            msgs.error("Not ready for this compound meta: ", meta_key)
```

Comment:
Change to:
```
msgs.error(f"Not ready for this compound meta: {meta_key}")
```
There is a second argument, but it should be a string with the type of class to use for the error.  If you copied this from somewhere else in the code, we should fix it there, too!